#162414180 controller, model and modelmanager for comment reaction

tosinibrahim96 opened the pull request:
**What does this PR do?**
Creates the model, Controller and model manager for comment reactions

**Description of Task to be completed?**
A user should be able to like a comment
A user should be able to dislike a comment
A user should be able to cancel their reaction 

**How should this be manually tested?**
clone this branch from github and run` npm test`

**What are the relevant pivotal tracker stories?**
[Pivotal Tracker Link](https://www.pivotaltracker.com/story/show/162414180)

Commit:
feat(likecomments): implement like functionality for comments
- Create like model with necessary relationships
- Feature has relationship with comments and users
[Delivers #162414180]

Commit:
[ft-#162414180] implement like and dislike feature
- A reaction can also be cancelled by the user
- Move strings to constant file
Delivers #162414180

Commit:
[ft-#162414180] implement like and dislike feature
- A reaction can also be cancelled by the user
Delivers #162414180

Commit:
[ft-#162414180] implement like and dislike feature
- A reaction can also be cancelled by the user
- Move strings to constant file
Delivers #162414180

codrex commented on constants/index.js:
```diff
@@ -41,15 +42,22 @@ const BAD_REQUEST_CODE = 400,
   RESET_LINK_SUBJECT = 'Reset Password - Authors Haven',
   BAD_TOKEN = '84ruyhf9q38uyg9',
+  COMMENT_NOT_FOUND = 'Comment does not exist',
+  REACTION_STATUS = 'You have successfully d this comment',
```
`You have successfully d this comment'` typo

tosinibrahim96 replied:
ooh, I am actually inserting a string between the text to format a message later for the user.

codrex replied:
ok

codrex commented on database/models/articlesComment.js:
```diff
@@ -1,5 +1,5 @@
 module.exports = (sequelize, DataTypes) => {
-  const articlesComment = sequelize.define('articlesComment', {
+  const articlesComments = sequelize.define('articlesComments', {
```
capitalize model names

tosinibrahim96 replied:
Will be changed

codrex commented on controllers/commentsReactionController.js:
```diff
@@ -0,0 +1,98 @@
+   * @returns {object| next} response | next function
+   * @memberof commentsReactionController
+   */
+  static async verifyCommentId(req, res, next) {
```
This should in the middleware folder

tosinibrahim96 replied:
done

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,114 @@
+   * @memberof commentsReactionModel
+   * @returns {uuid} The parameters from the query
+   */
+  static async confirmCommentId(id) {
```
`static async confirmCommentId(id) ` why is needed?

tosinibrahim96 replied:
It has been refactored

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,114 @@
+   * @returns {Boolean} reactionStatus
+   * @memberof commentsReactionModel
+   */
+  static async checkIfReactionExist(commentId, userId) {
```
This better off as a get reaction function.

tosinibrahim96 replied:
Done

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,114 @@
+      storedReaction = true;
+    } else {
+      storedReaction = false;
+    }
```
`
if (reaction === 'like') {
      storedReaction = true;
    } else {
      storedReaction = false;
    }
`
?

tosinibrahim96 replied:
It has been refactored

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,84 @@
+   *
+   *@param {uuid} id The id from request parameter
+   * @memberof commentsReactionModel
+   * @returns {uuid} The parameters from the query
```
` @returns {uuid} The parameters from the query` this function is returning a boolean due to the fact that you are performing an implicit typecasting here `return !!commentId;`

tosinibrahim96 replied:
i'll change that now

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,84 @@
+   * @returns {uuid} The parameters from the query
+   */
+  static async getComment(id) {
+    const commentId = await ArticlesComments.findOne({ where: { id } });
```
`findOne` in this case will be returning the full record object when the record is found. Why are you then assigning the returned record to a variable with the name `commentId`?

tosinibrahim96 replied:
I'll change that to comment now

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,84 @@
+   */
+  static async getComment(id) {
+    const commentId = await ArticlesComments.findOne({ where: { id } });
+    return !!commentId;
```
`return !!commentId;` why are you performing this typecasting?

tosinibrahim96 replied:
I'm trying to check if a comment exist from the controller, this returns true or false. I'lll remove the exclamations

tosinibrahim96 replied:
I just want to know if the comment already exist and i'm not really using the values. Just return true if it exists. That was my thought.

codrex commented on lib/modelManagers/commentsReactionModel.js:
```diff
@@ -0,0 +1,84 @@
+const { ArticlesComments, CommentsReactions } = require('../../database/models');
```
Why is `ArticlesComments` been imported here?

tosinibrahim96 replied:
The article comments model manager has not been merged and I have to check for a valid comment Id before a user can react.

tosinibrahim96 replied:
I need the model to do that

codrex replied:
When` article comments model manager` gets merged please raise a PR to fix this.

`const comment = await ArticlesComments.findOne({ where: { id } });`

tosinibrahim96 replied:
I'll do that

codrex commented on lib/utils/compareReactions.js:
```diff
@@ -0,0 +1,21 @@
+ * @returns {boolean} true or false
+ * @memberof commentsReactionModel
+ */
+const compareReactions = (newReaction, existingReaction) => {
```
Why is this still here?

tosinibrahim96 replied:
This should have been deleted. I'll remove this now

codrex commented on lib/utils/messageHandler.js:
```diff
@@ -1,16 +1,29 @@
+    success: UNSUCCESSFUL,
+    message
+  }),
+  formatReaction = reaction => [REACTION_STATUS.slice(0, 22), reaction, REACTION_STATUS.slice(22)].join('');
```
`formatReaction ` why is this here?

tosinibrahim96 replied:
This is a message handler file and you told me to move `formatreaction` away from the controller. I'm doing all the message formatting here

codrex commented on middlewares/valueVerifier.js:
```diff
@@ -0,0 +1,79 @@
+   */
+  static async verifyReactionId(req, res, next) {
+    try {
+      const reactionId = await CommentsReactions.findOne({
```
Is there no method in the CommentsReactions model manager to get a reaction?

tosinibrahim96 replied:
There is one that accepts different parameters. I'll refactor now

Commit:
[ft-#162414180] implement like and dislike feature

- A reaction can also be cancelled by the user
- Move strings to constant file
Delivers #162414180
